avro-converter/src/main/java/io/confluent/connect/avro/AvroData.java

```diff
@@ -795,13 +797,17 @@ private static Object maybeWrapSchemaless(Schema schema, Object value, String ty
               .map().values(fromConnectSchema(schema.valueSchema()));
         } else {
           // Special record name indicates format
+          final String entryTypeName = schema.name() == null ? MAP_ENTRY_TYPE_NAME : schema.name();
           org.apache.avro.SchemaBuilder.FieldAssembler<org.apache.avro.Schema> fieldAssembler
               = org.apache.avro.SchemaBuilder.builder()
               .array().items()
-              .record(MAP_ENTRY_TYPE_NAME).namespace(NAMESPACE).fields();
+              .record(entryTypeName).namespace(NAMESPACE).fields();
           addAvroRecordField(fieldAssembler, KEY_FIELD, schema.keySchema(), schemaMap);
           addAvroRecordField(fieldAssembler, VALUE_FIELD, schema.valueSchema(), schemaMap);
           baseSchema = fieldAssembler.endRecord();
+          if (schema.name() != null) {
+            baseSchema.getElementType().addProp(CONNECT_INTERNAL_TYPE_NAME, MAP_ENTRY_TYPE_NAME);
+          }
```

Comment:
this part looks different from what we did in #1036. might run into some hiccups when `pint merging`.